crowbar_framework/app/models/api/node.rb

```diff
@@ -158,7 +152,7 @@ def upgrade
 
       # this is just a fallback check, we should know by checking the global status that the action
       # should not be executed on already upgraded node
-      return true if upgraded?
+      return true if @node.file_exists? "/var/lib/crowbar/upgrade/node-upgraded-ok"
```

Comment:
This is not any `admin_upgrade` step, but all this is happening inside `nodes` step, as I wrote elsewhere. Indeed we'll need some API like this to check if particular sub-steps are finished, that's why the comment above this line says "this is just a fallback check"